### PR TITLE
Functional tests - Add new test 'BO > Product V2 > Check seo tab'

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add new test 'BO > Product V2 > Check seo tab'
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Ino need
| UI Tests          | Nightly run => https://github.com/nesrineabdmouleh/testing_pr/actions/runs/6417328861
| Fixed issue or discussion?     | no
| Related PRs       | no
| Sponsor company   |

### CODE REVIEW

File: tests/UI/pages/BO/catalog/productsV2/add/seoTab.ts [hunk 1]
```diff
@@ -2,6 +2,8 @@
 import BOBasePage from '@pages/BO/BObasePage';
 
 import type {Page} from 'playwright';
+import ProductData from "@data/faker/product";
+import createProductPage from "@pages/BO/catalog/productsV2/add/index";
```
Comment (Progi1984) [inline tests/UI/pages/BO/catalog/productsV2/add/seoTab.ts]:
```suggestion
import BOBasePage from '@pages/BO/BObasePage';
import createProductPage from '@pages/BO/catalog/productsV2/add/index';

import ProductData from '@data/faker/product';

import type {Page} from 'playwright';
```

File: tests/UI/campaigns/productV2/functional/18_seoTab.ts [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+import productsPage from '@pages/BO/catalog/productsV2';
+
+// Import FO pages
+import foProductPage from '@pages/FO/product';
+import {searchResultsPage} from '@pages/FO/searchResults';
+import categoryPageFO from '@pages/FO/category';
+
+// Import data
+import ProductData from '@data/faker/product';
+import Products from '@data/demo/products';
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/18_seoTab.ts]:
```suggestion
import Products from '@data/demo/products';
import ProductData from '@data/faker/product';
```

File: tests/UI/campaigns/productV2/functional/18_seoTab.ts [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+
+    it('should go to \'Catalog > Products\' page', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'goToProductsPage', baseContext);
+
+      await dashboardPage.goToSubMenu(
+        page,
+        dashboardPage.catalogParentLink,
+        dashboardPage.productsLink,
+      );
+
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/18_seoTab.ts]:
```suggestion
```

**Review by Progi1984 — COMMENTED** [review]
Small feedbacks

File: tests/UI/campaigns/productV2/functional/18_seoTab.ts (hunk 2 above)
Comment (nesrineabdmouleh) [reply tests/UI/campaigns/productV2/functional/18_seoTab.ts]: What's the problem ?

Comment (Progi1984) [reply tests/UI/campaigns/productV2/functional/18_seoTab.ts]: The order ;) 

We put import by alphabetic order in general.

**Review by Progi1984 — DISMISSED** [review]

**Comment by Progi1984** [comment]
@nesrineabdmouleh You must rebase.

**Review by Progi1984 — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

File: tests/UI/pages/BO/catalog/productsV2/add/seoTab.ts [hunk 4]
```diff
@@ -19,24 +20,122 @@ class SeoTab extends BOBasePage {
...
+  /**
+   * Get error message of friendly URL
+   * @param page {Page} Browser tab
+   * @returns {Promise<string>}
+   */
+  async getErrorMessageOfFriendlyUrl(page: Page): Promise<string> {
+    await this.clickAndWaitForLoadState(page, createProductPage.saveProductButton);
+
+    return this.getTextContent(page, this.alertText);
+  }
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/productsV2/add/seoTab.ts]: Maybe be having this in the addProductPage instead as is not linked to the this tab only

Comment (nesrineabdmouleh) [reply tests/UI/pages/BO/catalog/productsV2/add/seoTab.ts]: no it's an error alert related to friendlyUrl section

Capture d’écran de 2023-10-05 16-00-30

Comment (boubkerbribri) [reply tests/UI/pages/BO/catalog/productsV2/add/seoTab.ts]:
```suggestion
  /**
   * Get error message of friendly URL
   * @param page {Page} Browser tab
   * @returns {Promise<string>}
   */
  async getErrorMessageOfFriendlyUrl(page: Page): Promise<string> {
    return this.getTextContent(page, this.alertText);
  }
```

Maybe like this and do 2 actions on the test 1 for save the product the second to get the message

Comment (nesrineabdmouleh) [reply tests/UI/pages/BO/catalog/productsV2/add/seoTab.ts]: Done thx

**Review by Progi1984 — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Comment by boubkerbribri** [comment]
Thanks @nesrineabdmouleh

**Pull request merged**